File: tests/e2e/fixtures/code/sample.jl
# Julia: structs, multiple dispatch, comprehensions, broadcasting.
module Scanner

const LIMIT = 5 * 1024 * 1024

struct Note
    path::String
    tags::Vector{String}
    size::Int
end

Note(path; tags=String[]) = Note(path, tags, filesize(path))

islarge(n::Note) = n.size > LIMIT

function groupbytag(notes::AbstractVector{Note})
    out = Dict{String,Vector{Note}}()
    for n in notes, t in n.tags
        islarge(n) && continue
        push!(get!(out, t, Note[]), n)
    end
    out
end

describe(n::Integer) = n == 0 ? "no tags" : "$(n) tags"

end # module

File: tests/e2e/fixtures/code/sample.swift
// Swift: structs, protocols, optionals, closures, string interpolation.
import Foundation

let limit: UInt64 = 5 * 1024 * 1024

protocol Sized {
    var size: UInt64 { get }
}

struct Note: Sized {
    let path: String
    var tags: [String] = []

    var size: UInt64 {
        (try? FileManager.default.attributesOfItem(atPath: path)[.size] as? UInt64) ?? 0
    }
    var isLarge: Bool { size > limit }
}

func groupByTag(_ notes: [Note]) -> [String: [Note]] {
    var out: [String: [Note]] = [:]
    for n in notes where !n.isLarge {
        for t in n.tags { out[t, default: []].append(n) }
    }
    return out
}

let groups = groupByTag([Note(path: "a.md", tags: ["x", "y"])])
print("\(groups.count) tags")  // 2 tags
